src/function/vector_list_functions.cpp

```diff
@@ -634,6 +635,67 @@ std::unique_ptr<FunctionBindData> ListSumVectorFunction::bindFunc(
     return std::make_unique<FunctionBindData>(*resultType);
 }
 
+vector_function_definitions ListProductVectorFunction::getDefinitions() {
+    vector_function_definitions result;
+    result.push_back(std::make_unique<VectorFunctionDefinition>(LIST_PRODUCT_FUNC_NAME,
+        std::vector<LogicalTypeID>{LogicalTypeID::VAR_LIST}, LogicalTypeID::INT64, nullptr, nullptr,
+        bindFunc, false /* isVarlength*/));
+    return result;
+}
+
+std::unique_ptr<FunctionBindData> ListProductVectorFunction::bindFunc(
```

Comment:
I'm sure we should template this together with ListSum because their switch looks identical except the function itself is different.

As you do this refactor, check if other bindFunc can be templated.

Reply:
The error message thrown becomes `LIST_PRODUCT::bindFunc` instead of `ListProductVectorFunctions`